src/internal/operators/share.ts

```diff
@@ -84,30 +93,72 @@ export function share<T>(options: ShareConfig<T>): MonoTypeOperatorFunction<T>;
  * // ... and so on
  * ```
  *
+ * ## Example with notifier factory: Delayed reset
+ * ```ts
+ * import { interval } from 'rxjs';
+ * import { share, take, timer } from 'rxjs/operators';
+ *
+ * const source = interval(1000).pipe(take(3), share({ resetOnRefCountZero: () => timer(1000) }));
+ *
+ * const subscriptionOne = source.subscribe(x => console.log('subscription 1: ', x));
+ * setTimeout(() => subscriptionOne.unsubscribe(), 1300);
+ *
+ * setTimeout(() => source.subscribe(x => console.log('subscription 2: ', x)), 1700);
+ *
+ * setTimeout(() => source.subscribe(x => console.log('subscription 3: ', x)), 5000);
+ *
+ * // Logs:
+ * // subscription 1:  0
+ * // (subscription 1 unsubscribes here)
+ * // (subscription 2 subscribes here ~400ms later, source was not reset)
+ * // subscription 2:  1
+ * // subscription 2:  2
+ * // (subscription 2 unsubscribes here)
+ * // (subscription 3 subscribes here ~2000ms later, source did reset before)
+ * // subscription 3:  0
+ * // subscription 3:  1
+ * // subscription 3:  2
+ * ```
+ *
  * @see {@link api/index/function/interval}
  * @see {@link map}
  *
  * @return A function that returns an Observable that mirrors the source.
  */
-export function share<T>(options?: ShareConfig<T>): OperatorFunction<T, T> {
-  options = options || {};
-  const { connector = () => new Subject<T>(), resetOnComplete = true, resetOnError = true, resetOnRefCountZero = true } = options;
+export function share<T>(options: ShareConfig<T> = {}): MonoTypeOperatorFunction<T> {
```

Comment:
technically moving from `options = options || {}` to `options: ShareConfig<T> = {}` is a BC break, if you don't adhere to the typings. what is the stance towards this?

Reply:
I have no issue with this and I prefer the default that you've introduced. If someone is passing `null`, `0` or `""`, they've already violated our contract - expressed in the types and the docs that are derived from them - so if the behaviour of their code changes that's fine and and the change is non-breaking, IMO. Usage that violates the contract has undefined behaviour.